heat/core/_operations.py

```diff
@@ -38,11 +39,17 @@ def __binary_op(
         The operation to be performed. Function that performs operation elements-wise on the involved tensors,
         e.g. add values from other to self
     t1: DNDarray or scalar
-        The first operand involved in the operation,
+        The first operand involved in the operation.
     t2: DNDarray or scalar
-        The second operand involved in the operation,
+        The second operand involved in the operation.
     out: DNDarray, optional
-        Output buffer in which the result is placed
+        Output buffer in which the result is placed. If not provided, a freshly allocated array is returned.
+    where: DNDarray, optional
+        Condition to broadcast over the inputs. At locations where the condition is True, the `out` array
+        will be set to the result of the operation. Elsewhere, the `out` array will retain its original
+        value. If an uninitialized `out` array is created via the default `out=None`, locations within
+        it where the condition is False will remain uninitialized. If distributed, must be distributed
+        along the same dimension as the `out` array.
```

Comment:
This is only correct if `where` and `out` have the same shape. 

For example if `out` is (100, 10000) and distributed along the columns (`out.split` is 1), `where` is (10000,), their shapes are broadcastable but `where` must be distributed along 0.

Reply:
>"If distributed, the split axis (after broadcasting if required) must match that of the `out` array."

Would this phrasing be accurate?

Reply:
yes, sounds accurate